platform/darwin/src/local_glyph_rasterizer.mm

```diff
@@ -95,15 +154,63 @@ CTFontRef getFont() {
 LocalGlyphRasterizer::~LocalGlyphRasterizer()
 {}
 
+/**
+ Returns whether the rasterizer can rasterize a glyph for the given codepoint.
+ 
+ @param glyphID A font-agnostic Unicode codepoint, not a glyph index.
+ @returns Whether a glyph for the codepoint can be rasterized.
+ */
 bool LocalGlyphRasterizer::canRasterizeGlyph(const FontStack&, GlyphID glyphID) {
-    return util::i18n::allowsFixedWidthGlyphGeneration(glyphID) && impl->getFont();
+#if MBGL_DARWIN_NO_REMOTE_FONTS
+    return impl->isEnabled();
+#else
+    return util::i18n::allowsFixedWidthGlyphGeneration(glyphID) && impl->isEnabled();
+#endif
 }
 
-PremultipliedImage drawGlyphBitmap(GlyphID glyphID, CTFontRef font, Size size) {
-    PremultipliedImage rgbaBitmap(size);
-    
+/**
+ Draws the given codepoint into an image, gathers metrics about the glyph, and
+ returns the image.
+ 
+ @param glyphID A font-agnostic Unicode codepoint, not a glyph index.
+ @param font The font to apply to the codepoint.
+ @param metrics Upon return, the metrics match the font’s metrics for the glyph
+    representing the codepoint.
+ @returns An image containing the glyph.
+ */
+PremultipliedImage drawGlyphBitmap(GlyphID glyphID, CTFontRef font, GlyphMetrics& metrics) {
     CFStringRefHandle string(CFStringCreateWithCharacters(NULL, reinterpret_cast<UniChar*>(&glyphID), 1));
+    if (!string) {
+        throw std::runtime_error("Unable to create string from codepoint");
+    }
+
+    CFStringRef keys[] = { kCTFontAttributeName };
+    CFTypeRef values[] = { font };
+
+    CFDictionaryRefHandle attributes(
+        CFDictionaryCreate(kCFAllocatorDefault, (const void**)&keys,
+            (const void**)&values, sizeof(keys) / sizeof(keys[0]),
+            &kCFTypeDictionaryKeyCallBacks,
+            &kCFTypeDictionaryValueCallBacks));
+    if (!attributes) {
+        throw std::runtime_error("Unable to create attributed string attributes dictionary");
+    }
 
+    CFAttributedStringRefHandle attrString(CFAttributedStringCreate(kCFAllocatorDefault, *string, *attributes));
+    if (!attrString) {
+        throw std::runtime_error("Unable to create attributed string");
+    }
+    CTLineRefHandle line(CTLineCreateWithAttributedString(*attrString));
```

Comment:
This is still crashing despite the null check above, so that rules out an issue creating the attributed string and implicates a disabled font even more strongly:

```
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   com.apple.CoreText            	0x00007fff51ad3d82 TFontCascade::CreateFallback(__CTFont const*, __CFString const*, CTEmojiPolicy) const + 1194
1   com.apple.CoreText            	0x00007fff51ad2f7a TGlyphEncoder::AppendUnmappedCharRun(unsigned int, TCFRef<CTRun*>&, __CTFont const*, CFRange&, CFRange, TGlyphList<TDeletedGlyphIndex>&, TGlyphList<TDeletedGlyphIndex>&, TFontCascade const&, TGlyphEncoder::ClusterMatching, bool) + 460
2   com.apple.CoreText            	0x00007fff51ac9418 TGlyphEncoder::RunUnicodeEncoderRecursively(unsigned int, TCFRef<CTRun*>&&, __CTFont const*, CFRange, TGlyphList<TDeletedGlyphIndex>&, TGlyphList<TDeletedGlyphIndex>&, TFontCascade const*, TGlyphEncoder::ClusterMatching, bool) + 1306
3   com.apple.CoreText            	0x00007fff51ac8e84 TGlyphEncoder::RunUnicodeEncoder(TCFRef<CTRun*>&&, __CTFont const*, CFRange, TGlyphList<TDeletedGlyphIndex>&, TFontCascade const*) + 116
4   com.apple.CoreText            	0x00007fff51ac523a TGlyphEncoder::EncodeChars(CFRange, TAttributes const&, TGlyphEncoder::Fallbacks) + 1676
5   com.apple.CoreText            	0x00007fff51ac46e2 TTypesetterAttrString::Initialize(__CFAttributedString const*) + 236
6   com.apple.CoreText            	0x00007fff51ac4435 TTypesetterAttrString::TTypesetterAttrString(__CFAttributedString const*, __CFDictionary const*) + 163
7   com.apple.CoreText            	0x00007fff51ac4286 CTLineCreateWithAttributedString + 60
8   mbgl-test-runner              	0x000000010a5e88f2 mbgl::drawGlyphBitmap(char16_t, __CTFont const*, mbgl::GlyphMetrics&) + 2210 (local_glyph_rasterizer.mm:203)
```

Reply:
One thing I overlooked in https://github.com/mapbox/mapbox-gl-native/pull/16253#issuecomment-593683908 is that, on these CircleCI machines, the font descriptor keeps resolving to PingFangSC-Regular even though it doesn’t appear in the cascade list. I can’t get that to happen on my own machine.

<details>

```
static:build distiller$ plutil -p ~/Library/Preferences/com.apple.FontRegistry.user.plist
{
  "LastUpdated" => 2020-04-22 18:46:35 +0000
  "Version" => "18E226"
}
static:build distiller$ swift 
Welcome to Apple Swift version 5.1 (swiftlang-1100.0.270.13 clang-1100.0.33.7).
Type :help for assistance.
  1> import Foundation 
  2. import CoreText 
  3.  
  4. let attributes = [ 
  5. kCTFontSizeAttribute: 24, 
  6. kCTFontFamilyNameAttribute: "PingFang", 
  7. ] as [CFString : Any] 
  8. let desc = CTFontDescriptorCreateWithAttributes(attributes as CFDictionary) 
  9. let font = CTFontCreateWithFontDescriptor(desc, 0, nil) 
 10.  
 11. let attrs = [kCTFontAttributeName: font]
attributes: [CFString : Any] = 2 key/value pairs {
  [0] = {
    key = "NSFontFamilyAttribute"
    value = "PingFang"
  }
  [1] = {
    key = "NSFontSizeAttribute"
    value = 24
  }
}
desc: NSObject = {
  baseNSObject@0 = {
    isa = __NSCFType
  }
}
font: NSObject = {
  baseNSObject@0 = {
    isa = __NSCFType
  }
}
attrs: [CFString : CTFont] = 1 key/value pair {
  [0] = {
    key = "NSFont"
    value = {
      baseNSObject@0 = {
        isa = __NSCFType
      }
    }
  }
}
 12> let name = "PingFang TC"
name: String = "PingFang TC"
 13> let attributes = [ 
 14. kCTFontSizeAttribute: 24, 
 15. kCTFontFamilyNameAttribute: name, 
 16. kCTFontDisplayNameAttribute: name, 
 17. kCTFontNameAttribute: name, 
 18. ] as [CFString : Any]
attributes: [CFString : Any] = 4 key/value pairs {
  [0] = {
    key = "NSFontSizeAttribute"
    value = 24
  }
  [1] = {
    key = "NSFontNameAttribute"
    value = "PingFang TC"
  }
  [2] = {
    key = "NSFontFamilyAttribute"
    value = "PingFang TC"
  }
  [3] = {
    key = "NSFontVisibleNameAttribute"
    value = "PingFang TC"
  }
}
 19> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}1" as CFString, attributes as CFDictionary)
attrStr: CFAttributedString? = 0x0000000100211a90 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 20> print(CTLineCreateWithAttributedString(attrStr!))
<CTLine: 0x100254610>{run count = 2, string range = (0, 6), width = 66.6738, A/D/L = 12.72/4.08001/0, glyph count = 6, runs = (

<CTRun: 0x10012eec0>{string range = (0, 5), string = "\u8EAB\u4EC0\u6230\u30A2\u3042", attributes = <CFBasicHash 0x100400e30 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10012e830>{name = PingFangSC-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x100223dd0>{attributes = <CFBasicHash 0x10021f430 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10021d620 [0x7fffa7b378f0]>{contents = "PingFangSC-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100215640 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100215660 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100215680 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100253a20>{string range = (5, 1), string = "1", attributes = <CFBasicHash 0x100126560 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100215640 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100215660 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100215680 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 21> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\\\" as CFSt^CA2"+FFE" as CFString, attributes as CFDictionary) 
 21> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "A\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x00000001002564f0 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 22> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x100254860>{run count = 2, string range = (0, 6), width = 68.0039, A/D/L = 12.72/4.08001/0, glyph count = 6, runs = (

<CTRun: 0x1002549f0>{string range = (0, 1), string = "A", attributes = <CFBasicHash 0x100254820 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254700 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100254720 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100254740 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100254d50>{string range = (1, 5), string = "\u8EAB\u4EC0\u6230\u30A2\u3042", attributes = <CFBasicHash 0x1002578a0 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x100254b40>{name = PingFangSC-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x100223dd0>{attributes = <CFBasicHash 0x10021f430 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10021d620 [0x7fffa7b378f0]>{contents = "PingFangSC-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254700 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100254720 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100254740 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 23> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}A" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x0000000100254640 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 24> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x100257920>{run count = 2, string range = (0, 6), width = 68.0039, A/D/L = 12.72/4.08001/0, glyph count = 6, runs = (

<CTRun: 0x100254f20>{string range = (0, 5), string = "\u8EAB\u4EC0\u6230\u30A2\u3042", attributes = <CFBasicHash 0x100257ae0 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x100254d30>{name = PingFangSC-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x100223dd0>{attributes = <CFBasicHash 0x10021f430 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10021d620 [0x7fffa7b378f0]>{contents = "PingFangSC-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254a00 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100254a20 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100254a40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100255070>{string range = (5, 1), string = "A", attributes = <CFBasicHash 0x1002578e0 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254a00 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100254a20 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100254a40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 25> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\u{C548}\u{B155} \u{C138}\u{C0C1} KR" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x0000000100254b20 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 26> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x100408360>{run count = 4, string range = (0, 8), width = 64.8579, A/D/L = 10.8/3.60001/0, glyph count = 8, runs = (

<CTRun: 0x10013ee70>{string range = (0, 2), string = "\uC548\uB155", attributes = <CFBasicHash 0x100408560 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10013ec80>{name = AppleSDGothicNeo-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1002237b0>{attributes = <CFBasicHash 0x1002234a0 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x7fffa7bc3338 [0x7fffa7b378f0]>{contents = "AppleSDGothicNeo-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100257a00 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100257a20 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100257a40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100407f70>{string range = (2, 1), string = " ", attributes = <CFBasicHash 0x100135d30 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100257a00 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100257a20 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100257a40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x1004080c0>{string range = (3, 2), string = "\uC138\uC0C1", attributes = <CFBasicHash 0x1004084f0 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10013ec80>{name = AppleSDGothicNeo-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1002237b0>{attributes = <CFBasicHash 0x1002234a0 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x7fffa7bc3338 [0x7fffa7b378f0]>{contents = "AppleSDGothicNeo-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100257a00 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100257a20 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100257a40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100408210>{string range = (5, 3), string = " KR", attributes = <CFBasicHash 0x100135d30 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100257a00 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100257a20 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100257a40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 27> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}5" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x0000000100138590 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 28> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x10012eec0>{run count = 2, string range = (0, 6), width = 66.6738, A/D/L = 12.72/4.08001/0, glyph count = 6, runs = (

<CTRun: 0x10013ee90>{string range = (0, 5), string = "\u8EAB\u4EC0\u6230\u30A2\u3042", attributes = <CFBasicHash 0x100103330 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x100145850>{name = PingFangSC-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x100223dd0>{attributes = <CFBasicHash 0x10021f430 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10021d620 [0x7fffa7b378f0]>{contents = "PingFangSC-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10013a0f0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x10013a2b0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x10013a8b0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100145a40>{string range = (5, 1), string = "5", attributes = <CFBasicHash 0x100103280 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10013a0f0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x10013a2b0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x10013a8b0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 29> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "B\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x00000001002573b0 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 30> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x100103280>{run count = 2, string range = (0, 6), width = 68.0039, A/D/L = 12.72/4.08001/0, glyph count = 6, runs = (

<CTRun: 0x100138cd0>{string range = (0, 1), string = "B", attributes = <CFBasicHash 0x10013a490 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254840 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x1002578d0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x1002578f0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100138120>{string range = (1, 5), string = "\u8EAB\u4EC0\u6230\u30A2\u3042", attributes = <CFBasicHash 0x10013a310 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10012e830>{name = PingFangSC-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x100223dd0>{attributes = <CFBasicHash 0x10021f430 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10021d620 [0x7fffa7b378f0]>{contents = "PingFangSC-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254840 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x1002578d0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x1002578f0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 31> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}S2" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x000000010023ea80 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 32> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x100103280>{run count = 2, string range = (0, 7), width = 74.6777, A/D/L = 12.72/4.08001/0, glyph count = 7, runs = (

<CTRun: 0x100138cd0>{string range = (0, 5), string = "\u8EAB\u4EC0\u6230\u30A2\u3042", attributes = <CFBasicHash 0x10013a370 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10012e830>{name = PingFangSC-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x100223dd0>{attributes = <CFBasicHash 0x10021f430 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x10021d620 [0x7fffa7b378f0]>{contents = "PingFangSC-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254d40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100254d60 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100254d80 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x10012eec0>{string range = (5, 2), string = "S2", attributes = <CFBasicHash 0x10013a8d0 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100254d40 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100254d60 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x100254d80 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 33> let attrStr = CFAttributedStringCreate(kCFAllocatorDefault, "\u{C548}\u{B155} \u{C138}\u{C0C1}8" as CFString, attributes as CFDictionary) 
attrStr: CFAttributedString? = 0x000000010013eb30 {
  baseNSMutableAttributedString@0 = {
    baseNSAttributedString@0 = {
      baseNSObject@0 = {
        isa = __NSCFAttributedString
      }
    }
  }
}
 34> print(CTLineCreateWithAttributedString(attrStr!)) 
<CTLine: 0x100103280>{run count = 4, string range = (0, 6), width = 51.5278, A/D/L = 10.8/3.60001/0, glyph count = 6, runs = (

<CTRun: 0x100138cd0>{string range = (0, 2), string = "\uC548\uB155", attributes = <CFBasicHash 0x10013bb10 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10012e830>{name = AppleSDGothicNeo-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1002237b0>{attributes = <CFBasicHash 0x1002234a0 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x7fffa7bc3338 [0x7fffa7b378f0]>{contents = "AppleSDGothicNeo-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100136440 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100138790 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x1001380a0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100138120>{string range = (2, 1), string = " ", attributes = <CFBasicHash 0x10013a610 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100136440 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100138790 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x1001380a0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100138810>{string range = (3, 2), string = "\uC138\uC0C1", attributes = <CFBasicHash 0x100135d30 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x10012e830>{name = AppleSDGothicNeo-Regular, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1002237b0>{attributes = <CFBasicHash 0x1002234a0 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x7fffa7bc3338 [0x7fffa7b378f0]>{contents = "AppleSDGothicNeo-Regular"}
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100136440 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100138790 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x1001380a0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}


<CTRun: 0x100138ec0>{string range = (5, 1), string = "8", attributes = <CFBasicHash 0x10013a610 [0x7fffa7b378f0]>{type = mutable dict, count = 5,
entries =>
	0 : <CFString 0x7fffa7eef3a0 [0x7fffa7b378f0]>{contents = "NSFont"} = <CTFont: 0x1001265a0>{name = Helvetica, size = 12.000000, matrix = 0x0, descriptor = <CTFontDescriptor: 0x1001239c0>{attributes = <CFBasicHash 0x100123940 [0x7fffa7b378f0]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = Helvetica
}
>}}
	1 : <CFString 0x7fffa7efe820 [0x7fffa7b378f0]>{contents = "NSFontSizeAttribute"} = <CFNumber 0x752d850abac46a3f [0x7fffa7b378f0]>{value = +24, type = kCFNumberSInt64Type}
	3 : <CFString 0x7fffa7efe580 [0x7fffa7b378f0]>{contents = "NSFontNameAttribute"} = <CFString 0x100136440 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	5 : <CFString 0x7fffa7efe5e0 [0x7fffa7b378f0]>{contents = "NSFontFamilyAttribute"} = <CFString 0x100138790 [0x7fffa7b378f0]>{contents = "PingFang TC"}
	6 : <CFString 0x7fffa7efe5c0 [0x7fffa7b378f0]>{contents = "NSFontVisibleNameAttribute"} = <CFString 0x1001380a0 [0x7fffa7b378f0]>{contents = "PingFang TC"}
}
}

)
}
 35> Connection to 162.252.208.238 closed by remote host.
Connection to 162.252.208.238 closed.
```

</details>

Reply:
Running `system_profiler SPFontsDataType` shows the relevant fonts are activated:

```
        PingFangSC-Regular:
          Full Name: PingFang SC Regular
          Family: PingFang SC
          Style: Regular
          Version: 13.0d1e3
          Vendor: DynaComware Shanghai Limited; DynaComware Hong Kong Limited; DynaComware Taiwan Inc.
          Unique Name: PingFang SC Regular; 13.0d1e3; 2017-06-09
          Copyright: Copyright © 2015 DynaComware. All rights reserved.
          Trademark: PingFang is a trademark of Apple Inc.
          Description: Designed by DynaComware & Apple.
          Outline: Yes
          Valid: Yes
          Enabled: Yes
          Duplicate: No
          Copy Protected: No
          Embeddable: Yes
        PingFangTC-Regular:
          Full Name: PingFang TC Regular
          Family: PingFang TC
          Style: Regular
          Version: 13.0d1e3
          Vendor: DynaComware Shanghai Limited; DynaComware Hong Kong Limited; DynaComware Taiwan Inc.
          Unique Name: PingFang TC Regular; 13.0d1e3; 2017-06-09
          Copyright: Copyright © 2015 DynaComware. All rights reserved.
          Trademark: PingFang is a trademark of Apple Inc.
          Description: Designed by DynaComware & Apple.
          Outline: Yes
          Valid: Yes
          Enabled: Yes
          Duplicate: No
          Copy Protected: No
          Embeddable: Yes
        AppleSDGothicNeo-Regular:
          Full Name: Apple SD Gothic Neo Regular
          Family: Apple SD Gothic Neo
          Style: Regular
          Version: 13.0d1e9
          Vendor: Sandoll Communications Inc.
          Unique Name: Apple SD Gothic Neo Regular; 13.0d1e9; 2017-06-14
          Designer: Sandoll Communications Inc.
          Copyright: Copyright © 2014 Sandoll Communications Inc. All rights reserved.
          Trademark: SDGothicNeo is a registered trademark of Sandoll Communications Inc.
          Outline: Yes
          Valid: Yes
          Enabled: Yes
          Duplicate: No
          Copy Protected: No
          Embeddable: Yes
```

Reply:
I managed to reproduce the crash on the build machine with the following code:

```swift
import Foundation
import CoreText

let fontDescriptors = [
    CTFontDescriptorCreateWithAttributes([
        kCTFontSizeAttribute: 24,
        kCTFontNameAttribute: "NotoCJK",
        kCTFontDisplayNameAttribute: "NotoCJK",
        kCTFontFamilyNameAttribute: "NotoCJK",
    ] as CFDictionary),
    CTFontDescriptorCreateWithAttributes([
        kCTFontSizeAttribute: 24,
        kCTFontNameAttribute: "PingFang TC",
        kCTFontDisplayNameAttribute: "PingFang TC",
        kCTFontFamilyNameAttribute: "PingFang TC",
    ] as CFDictionary),
]
let attributes: [CFString: Any] = [
    kCTFontSizeAttribute: 24,
    kCTFontCascadeListAttribute: fontDescriptors,
]
let desc = CTFontDescriptorCreateCopyWithAttributes(fontDescriptors[0], attributes as CFDictionary)
let font = CTFontCreateWithFontDescriptor(desc, 0, nil)
let glyphs = "\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}1\nA\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}\n\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}A\n\u{C548}\u{B155} \u{C138}\u{C0C1} KR\n\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}5\nB\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}\n\u{8EAB}\u{4EC0}\u{6230}\u{30A2}\u{3042}S2\n\u{C548}\u{B155} \u{C138}\u{C0C1}8"
let strings = glyphs.utf16.map { (glyph) -> CFString in
    var glyphID = UniChar(glyph)
    return CFStringCreateWithCharacters(nil, &glyphID, 1)
}
let attrStrs = strings.map { CFAttributedStringCreate(kCFAllocatorDefault, $0, [kCTFontAttributeName: font] as CFDictionary) }
CTLineCreateWithAttributedString(attrStrs[0]!)
```

```
Execution interrupted. Enter code to recover and continue.
Enter LLDB commands to investigate (type :help for assistance.)
Process 8244 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x00007fff39defd82 CoreText`TFontCascade::CreateFallback(__CTFont const*, __CFString const*, CTEmojiPolicy) const + 1194
CoreText`TFontCascade::CreateFallback:
->  0x7fff39defd82 <+1194>: movq   (%rdi), %rax
    0x7fff39defd85 <+1197>: callq  *0x2b8(%rax)
    0x7fff39defd8b <+1203>: testb  %al, %al
    0x7fff39defd8d <+1205>: movq   -0x38(%rbp), %rbx
Target 0: (repl_swift) stopped.
```

The crash seems to be from reusing the same font descriptor as the main font descriptor but keeping it in the cascade list. Effectively the font descriptor is part of its own cascade list. Pulling the first font descriptor out of the cascade list seems to fix the issue, at least in this environment.